tslib.es6.js

```diff
@@ -209,13 +209,20 @@ var __setModuleDefault = Object.create ? (function(o, v) {
     o["default"] = v;
 };
 
-export function __importStar(mod) {
-    if (mod && mod.__esModule) return mod;
-    var result = {};
-    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
-    __setModuleDefault(result, mod);
-    return result;
-}
+export var __importStar = function (mod) {
+    var cache = typeof WeakMap === "function" ? new WeakMap() : null;

```

Comment:
Are you proposing this be implemented in TypeScript as well for inline helper emit? If so we'd need to introduce a checker error if you declare your own value named `WeakMap` at the top of a module.

Reply:
The issue there is that this is supposed to be an optimisation when two different modules that use `tslib` import the same **CommonJS** module, the value returned by `__importStar` is the same value for both imports, e.g.:

```js
// modA.js
Object.defineProperty(exports, "__esModule", { value: true });
var tslib = require("tslib");

exports.example = tslib.__importStar(require("example"));
```
```js
// modB.js
Object.defineProperty(exports, "__esModule", { value: true });
var tslib = require("tslib");

exports.example = tslib.__importStar(require("example"));
```
```js
// test.js
var assert = require("assert");

var modA = require("./modA.js");
var modB = require("./modA.js");

// The following passes regardless of whether `example` has `__esModule` or not:
assert.strictEqual(modA.example, modB.example);
```

---

It doesn’t make sense to include the module cache in inline helpers, as `modA` and `modB` would then each have their own cache, and every new local `__imporStar` call would then have a cache miss, since the cache wouldn’t be global.